tests/scripts/create_coverage_report.py

```diff
@@ -14,40 +13,33 @@
 LOGGER = logging.getLogger(__name__)
 
 
-def merge_all_info_files(start_path, merge_file_name):
+def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
+    path_to_info_files = os.environ["TMT_TREE"].split('/tree')[0] + "/execute/data/"
+    path_to_tests_results = os.environ["TMT_TREE"].split('/tree')[0] + "/report/default-0"
+    merge_file_name = "merged.info"
+    merge_dir = "/tmp"
+    report_dir_name = "/report"
+
+    # Copy info files to the ctrl container and run lcov command
+    root = pathlib.Path(path_to_info_files)
     lcov_list = list()
     lcov_list.append("lcov")
-
-    root = pathlib.Path(start_path)
-
     for file_path in root.glob("**/*.info"):
         file = str(file_path)
+        file_name = file.split("/")[-1]
         LOGGER.debug(f"Found info file '{file}'")
+        content = read_file(file_path)
+        ctrl.create_file(merge_dir, file_name, content)
         lcov_list.append("-a")
-        lcov_list.append(file)
+        lcov_list.append(f"{merge_dir}/{file_name}")
 
     lcov_list.append("-o")
-    lcov_list.append(merge_file_name)
-
-    command.Command(args=" ".join(lcov_list)).run()
-
-
-def exec(ctrl: BluechiControllerContainer, nodes: Dict[str, BluechiNodeContainer]):
-    path_to_info_files = os.environ["TMT_TREE"].split('/tree')[0] + "/execute/data/"
-    path_to_tests_results = os.environ["TMT_TREE"].split('/tree')[0] + "/report/default-0"
-    merge_file_name = "merged.info"
-    merge_file_full_path = path_to_tests_results + "/" + merge_file_name
-    merge_dir = "/tmp"
-    report_dir_name = "/report"
-
-    merge_all_info_files(path_to_info_files, merge_file_full_path)
+    lcov_list.append(f"{merge_dir}/{merge_file_name}")
 
-    if os.path.exists(merge_file_full_path):
-        LOGGER.debug(f"Generating report for merged info file '{merge_file_full_path}'")
-        content = read_file(merge_file_full_path)
-        ctrl.create_file(merge_dir, merge_file_name, content)
-        ctrl.exec_run(f"genhtml {merge_dir}/{merge_file_name} --output-directory={report_dir_name}")
-        ctrl.get_file(report_dir_name, path_to_tests_results)
+    ctrl.exec_run(" ".join(lcov_list))
+    ctrl.exec_run(f"genhtml {merge_dir}/{merge_file_name} --output-directory={report_dir_name}")
```

Comment:
Could you add checks for the results in the form of
```python
result, output = ctrl.exec(...)
if result != 0:
   raise Exception(f"Failed to do xyz: {output}")
```
This makes debugging the coverage collection easier in case something fails here.